### PR TITLE
add purchase prices

### PR DESCRIPTION
**Pull request by patrickweh**

### CODE REVIEW

File: lang/de.json [hunk 1]
```diff
@@ -246,6 +250,7 @@
     "After amount of events": "Nach Anzahl an Terminen",
     "After number of recurrences": "Nach Anzahl an Wiederholungen",
     "Agent": "Vertreter",
+    "Agent Id": "Agenten-ID",
```
Comment (SirSplasch) [inline lang/de.json:253]: Vertreter-ID

File: lang/de.json [hunk 2]
```diff
@@ -258,13 +263,16 @@
     "all-day": "ganzen Tag",
     "Amount": "Anzahl",
     "amount": "Anzahl",
+    "Amount Bundle": "Bündelmenge",
+    "Amount Packed Products": "Menge verpackter Produkte",
```
Comment (SirSplasch) [inline lang/de.json:267]: Anzahl verpackter Produkte

File: lang/de.json [hunk 3]
```diff
@@ -258,13 +263,16 @@
...
     "Ancestors": "Vorgänger",
     "and": "und",
     "Angle": "Winkel",
     "Answer": "Antworten",
     "Answer to customer": "Antwort an Kunden",
     "Append links": "Links hinzufügen",
     "Apply": "Anwenden",
+    "Approval User Id": "Genehmigungsbenutzer-ID",
```
Comment (SirSplasch) [inline lang/de.json:275]: ??? More like Freigabe-Benutzer-ID?

File: lang/de.json [hunk 4]
```diff
@@ -292,34 +302,48 @@
     "Bank Code": "Banknummer",
     "Bank Connection": "Bankverbindung",
     "Bank connection": "Bankverbindung",
+    "Bank Connection Id": "Bankverbindung-ID",
     "Bank Connections": "Bankverbindungen",
     "Bank connections": "Bankverbindungen",
     "Bank Name": "Bankname",
+    "Basic Unit": "Basiseinheit",
```
Comment (SirSplasch) [inline lang/de.json:309]: Grundeinheit

File: lang/de.json [hunk 5]
```diff
@@ -292,34 +302,48 @@
...
     "Cart saved to watchlist": "Warenkorb in Merkliste gespeichert",
     "Categories": "Kategorien",
     "Category": "Kategorie",
+    "Category Id": "Kategorie-ID",
     "Causer": "Verursacher",
     "Causer Id": "Verursacher-ID",
     "Causer Type": "Verursacher-Typ",
+    "Cc": "CC",
+    "Ceo": "Geschäftsführer",
     "CEO": "Geschäftführer",
```
Comment (SirSplasch) [inline lang/de.json:344]: Geschäftsführer

File: lang/de.json [hunk 6]
```diff
@@ -335,10 +359,12 @@
     "Click to login": "Klicken zum Anmelden",
     "Client": "Mandant",
     "Client Code": "Mandantennummer",
+    "Client Id": "Kunden-ID",
```
Comment (SirSplasch) [inline lang/de.json:362]: Mandanten-ID

File: lang/de.json [hunk 7]
```diff
@@ -361,10 +390,13 @@
     "Company": "Firma",
     "company": "Firma",
     "Confirm Delete Event": "Termin löschen",
+    "Confirm Option": "Bestätigungsoption",
     "Confirmed": "Bestätigt",
     "Contact": "Kontakt",
     "Contact Bank Connection": "Bankverbindung",
+    "Contact Bank Connection Id": "Kontaktbankverbindung-ID",
```
Comment (SirSplasch) [inline lang/de.json:397]: Bankverbindung-ID

File: lang/de.json [hunk 8]
```diff
@@ -374,14 +406,21 @@
...
+    "Counterpart Account Number": "Gegenpartei Kontonummer",
+    "Counterpart Bank Name": "Gegenpartei Bankname",
     "Counterpart BAnk Name": "Gegenpartei Bank Name",
+    "Counterpart Bic": "Gegenpartei BIC",
+    "Counterpart Iban": "Gegenpartei IBAN",
     "Counterpart IBAN": "Gegenpartei IBAN",
     "Counterpart Name": "Gegenpartei Name",
     "Countries": "Länder",
     "Country": "Land",
+    "Country Id": "Länder-ID",
```
Comment (SirSplasch) [inline lang/de.json:421]: Land-ID

File: lang/de.json [hunk 9]
```diff
@@ -405,33 +444,44 @@
     "Create User": "Benutzer anlegen",
     "created": "hat erstellt",
     "Created At": "Erstellt am",
+    "Created By": "Erstellt von",
     "Credit Limit": "Kreditlimit",
+    "Credit Line": "Kreditlinie",
     "Credit line": "Kreditlinie",
     "Creditor Identifier": "Gläubiger-Identifikationsnummer",
+    "Creditor Number": "Gläubigernummer",
     "Creditor number": "Kreditor-Nummer",
```
Comment (SirSplasch) [inline lang/de.json:453]: decide whether it should be translated to "Gläubigernummer" or "Kreditor-Nummer"

File: lang/de.json [hunk 10]
```diff
@@ -405,33 +444,44 @@
...
+    "Date Of Approval": "Genehmigungsdatum",
     "Date Of Birth": "Geburtsdatum",
     "Date, Location and Signature": "Datum, Ort und Unterschrift",
     "Day": "Tag",
     "Day Constraints": "Tages-Einschränkungen",
     "Day(s)": "Tag(e)",
     "DayGridWeek": "Tagesraster Woche",
     "Days": "Tage",
+    "Debtor Number": "Schuldnernummer",
     "Debtor number": "Debitor-Nummer",
```
Comment (SirSplasch) [inline lang/de.json:485]: decide whether it should be "Schuldnernummer" or "Debitor-Nummer"

File: lang/de.json [hunk 11]
```diff
@@ -405,33 +444,44 @@
...
     "Customer": "Kunde",
+    "Customer Delivery Date": "Kundenlieferdatum",
     "Customer no.": "Kunden-Nr.",
     "Customer Number": "Kundennummer",
     "Customer Portal": "Kundenportal",
     "Customer portal": "Kundenportal",
     "Daily": "Täglich",
     "Daily At": "Täglich um",
     "Dashboard module": "Dashboard Modul",
+    "Datanorm Long Text": "Datanorm Langtext",
```
Comment (SirSplasch) [inline lang/de.json:473]: remove

File: lang/de.json [hunk 12]
```diff
@@ -571,16 +634,21 @@
     "Every Two Hours": "Alle zwei Stunden",
     "Every Two Minutes": "Alle zwei Minuten",
     "Every Two Seconds": "Alle zwei Sekunden",
+    "Excluded": "Ausgeschlossen",
     "Execute": "Ausführen",
     "Execute Payment Run": "Zahlungslauf ausführen",
     "Execution Date": "Ausführungsdatum",
+    "Expense Ledger Account Id": "Kostenkontenplan-ID",
```
Comment (SirSplasch) [inline lang/de.json:641]: I'm unsure whether this translation is right.

File: lang/de.json [hunk 13]
```diff
@@ -603,10 +674,21 @@
     "General": "Allgemein",
     "German": "Deutsch",
     "Get new messages": "Neue Nachrichten abrufen",
+    "Give": "Geben",
     "Gram": "Gramm",
     "gross": "brutto",
+    "Gross Profit": "Rohertrag",
```
Comment (SirSplasch) [inline lang/de.json:680]: "Rohertrag"???

File: lang/de.json [hunk 14]
```diff
@@ -603,10 +674,21 @@
...
     "Group": "Gruppe",
+    "Guard Name": "Guard-Name",
+    "Has Delivery Lock": "Hat Liefersperre",
+    "Has Logistic Notify Number": "Logistische Benachrichtigungsnummer",
+    "Has Logistic Notify Phone Number": "Logistische Benachrichtigungsnummer",
```
Comment (SirSplasch) [inline lang/de.json:685]: please re-translate this, this can't be right.

File: lang/de.json [hunk 15]
```diff
@@ -603,10 +674,21 @@
...
     "German": "Deutsch",
     "Get new messages": "Neue Nachrichten abrufen",
+    "Give": "Geben",
     "Gram": "Gramm",
     "gross": "brutto",
+    "Gross Profit": "Rohertrag",
     "Group": "Gruppe",
+    "Guard Name": "Guard-Name",
+    "Has Delivery Lock": "Hat Liefersperre",
+    "Has Logistic Notify Number": "Logistische Benachrichtigungsnummer",
```
Comment (SirSplasch) [inline lang/de.json:684]: please re-translate this, this can't be right.

File: lang/de.json [hunk 16]
```diff
@@ -603,10 +674,21 @@
...
+    "Has Logistic Notify Phone Number": "Logistische Benachrichtigungsnummer",
+    "Has Repeatable Events": "Hat wiederholbare Ereignisse",
     "Has repeatable events": "Wiederholbare Termine",
+    "Has Sensitive Reminder": "Hat sensible Erinnerung",
```
Comment (SirSplasch) [inline lang/de.json:688]: This translation seems a bit weird.

File: lang/de.json [hunk 17]
```diff
@@ -626,8 +709,12 @@
     "hours": "Stunden",
     "Hours:Minutes": "Stunden:Minuten",
     "Hover menu item": "Gehoverter Navigationspunkt",
+    "Html": "HTML",
+    "Html Body": "HTML-Körper",
```
Comment (SirSplasch) [inline lang/de.json:713]: more like "HTML-Inhalt"

File: lang/de.json [hunk 18]
```diff
@@ -652,38 +742,79 @@
...
     "Is Instant Payment": "Ist Sofortüberweisung",
+    "Is Internal": "Intern",
+    "Is Invoice Address": "Rechnungsadresse",
     "Is Locked": "Ist gesperrt",
     "Is Main Address": "Ist Hauptadresse",
+    "Is Merge Invoice": "Rechnung zusammenführen",
     "Is Net": "Netto",
+    "Is New Customer": "Neukunde",
+    "Is Nos": "NOS",
     "Is NOS": "Ist NOS",
```
Comment (SirSplasch) [inline lang/de.json:780]: "NOS" or "Ist NOS" keep it consistent.

File: lang/de.json [hunk 19]
```diff
@@ -652,38 +742,79 @@
...
     "is not null": "Ist nicht leer",
+    "Is Notifiable": "Benachrichtigbar",
     "is null": "Ist leer",
+    "Is O Auth": "OAuth",
```
Comment (SirSplasch) [inline lang/de.json:784]: does this even exist?

File: lang/de.json [hunk 20]
```diff
@@ -652,38 +742,79 @@
...
+    "Is Paid": "Bezahlt",
     "Is Pause": "Ist Pause",
     "Is Percentage": "Prozent",
+    "Is Portal Public": "Portal Öffentlich",
+    "Is Pre Filled": "Vorgefüllt",
+    "Is Product Serial Number": "Produkt-Seriennummer",
+    "Is Public": "Öffentlich",
     "Is Purchase": "Ist Einkauf",
+    "Is Required Manufacturer Serial Number": "Hersteller-Seriennummer erforderlich",
+    "Is Required Product Serial Number": "Produkt-Seriennummer erforderlich",
```
Comment (SirSplasch) [inline lang/de.json:794]: does this even exist?

File: lang/de.json [hunk 21]
```diff
@@ -652,38 +742,79 @@
...
+    "Is O Auth": "OAuth",
+    "Is Paid": "Bezahlt",
     "Is Pause": "Ist Pause",
     "Is Percentage": "Prozent",
+    "Is Portal Public": "Portal Öffentlich",
+    "Is Pre Filled": "Vorgefüllt",
+    "Is Product Serial Number": "Produkt-Seriennummer",
+    "Is Public": "Öffentlich",
     "Is Purchase": "Ist Einkauf",
+    "Is Required Manufacturer Serial Number": "Hersteller-Seriennummer erforderlich",
```
Comment (SirSplasch) [inline lang/de.json:793]: does this even exist?

File: lang/de.json [hunk 22]
```diff
@@ -694,9 +825,12 @@
     "Last Year": "Letztes Jahr",
     "Lastname": "Nachname",
     "lastname": "Nachname",
+    "Latitude": "Breitengrad",
     "Lay out user": "Auslage Benutzer",
+    "Lay Out User Id": "Benutzer-ID",
```
Comment (SirSplasch) [inline lang/de.json:830]: Auslage Benutzer-ID

File: lang/de.json [hunk 23]
```diff
@@ -694,9 +825,12 @@
...
     "Leave empty to generate a new :attribute.": "Leer lassen, um eine neue :attribute zu erzeugen.",
     "Ledger Account": "Buchungskonto",
+    "Ledger Account Id": "Kontenplan-ID",
```
Comment (SirSplasch) [inline lang/de.json:833]: Buchungskonto-ID

File: lang/de.json [hunk 24]
```diff
@@ -715,38 +849,59 @@
     "Login failed": "Anmeldung fehlgeschlagen",
     "Login link sent, check your inbox": "Anmelde-Link gesendet, bitte überprüfen Sie Ihr Postfach",
     "Login name": "Login-name",
+    "Logistic Note": "Logistische Notiz",
```
Comment (SirSplasch) [inline lang/de.json:852]: Logistik Notiz

File: lang/de.json [hunk 25]
```diff
@@ -715,38 +849,59 @@
...
+    "Mailbox Zip": "Postfach PLZ",
     "Main address": "Hauptadresse",
     "Main Address": "Hauptadresse",
     "Main product files": "Hauptproduktdateien",
     "Manage discount group": "Rabattgruppe verwalten",
     "Manage Logos": "Logos verwalten",
     "Manage users": "Benutzer verwalten",
     "Mandate Reference Number": "Mandatsreferenz",
+    "Manufacturer Product Number": "Hersteller-Produktnummer",
     "Manufacturer product number": "Herstellerproduktnummer",
```
Comment (SirSplasch) [inline lang/de.json:879]: keep these two consistent. Although I don't think this even exists, can probably be removed.

File: lang/de.json [hunk 26]
```diff
@@ -715,38 +849,59 @@
...
     "Margin": "Marge",
     "Mark all as read": "Alle als gelesen markieren",
     "Mark found as done": "Gefundene als abgeschlossen markieren",
     "Max Amount": "Maximalbetrag",
     "Max Booking Date": "Maximales Buchungsdatum",
+    "Max Delivery Time": "Maximale Lieferzeit",
+    "Max Purchase": "Maximalkauf",
```
Comment (SirSplasch) [inline lang/de.json:886]: "Maximalkauf"?

File: lang/de.json [hunk 27]
```diff
@@ -715,38 +849,59 @@
...
     "media.{media}.{filename}.get": "Medien anzeigen",
     "Message": "Nachricht",
+    "Message Id": "Nachrichten-ID",
+    "Message Uid": "Nachrichten-UID",
+    "Migrate": "Migrieren",
+    "Mime Type": "MIME-Typ",
     "Min Amount": "Mindestbetrag",
     "Min Booking Date": "Mindestbuchungsdatum",
+    "Min Delivery Time": "Minimale Lieferzeit",
+    "Min Purchase": "Mindestkauf",
```
Comment (SirSplasch) [inline lang/de.json:900]: "Mindestkauf"?

File: lang/de.json [hunk 28]
```diff
@@ -820,52 +975,69 @@
...
     "Only order positions from given order allowed.": "Nur Aufragspositionen des gegebenen Auftrags sind erlaubt.",
     "Open": "Offen",
     "open": "offen",
+    "Opening Hours": "Öffnungszeiten",
     "Open…": "Öffnen…",
+    "Options": "Optionen",
     "or": "oder",
     "Order": "Auftrag",
     "Order by": "Sortieren nach",
+    "Order Column": "Bestellspalte",
```
Comment (SirSplasch) [inline lang/de.json:997]: "Sortierspalte"?

File: lang/de.json [hunk 29]
```diff
@@ -820,52 +975,69 @@
...
     "Order Confirmation": "Auftragsbestätigung",
     "Order Confirmation for order :order_number": "Auftragsbestätigung für Auftrag :order_number",
     "Order Date": "Auftragsdatum",
     "Order date": "Auftragsdatum",
     "Order details": "Auftragsdetails",
+    "Order Id": "Bestell-ID",
```
Comment (SirSplasch) [inline lang/de.json:1003]: "Auftrag-ID"

File: lang/de.json [hunk 30]
```diff
@@ -820,52 +975,69 @@
...
     "Order no.": "Auftrags-Nr.",
     "Order Number": "Auftragsnummer",
     "Order Position": "Auftragsposition",
     "Order position does not exists in parent order.": "Auftragsposition existiert nicht im Hauptauftrag",
+    "Order Position Id": "Bestellposition-ID",
```
Comment (SirSplasch) [inline lang/de.json:1008]: "Auftragsposition-ID"

File: lang/de.json [hunk 31]
```diff
@@ -820,52 +975,69 @@
...
     "Order Positions": "Auftragspositionen",
     "Order positions": "Auftragspositionen",
+    "Order saved successfully!": "Auftrag erfolgreich gespeichert!",
     "Order State": "Auftragsstatus",
     "Order state": "Auftragsstatus",
     "Order Type": "Auftragsart",
     "Order type": "Auftragsart",
+    "Order Type Enum": "Bestelltyp",
```
Comment (SirSplasch) [inline lang/de.json:1016]: "Auftragstyp"

File: lang/de.json [hunk 32]
```diff
@@ -820,52 +975,69 @@
...
+    "Order Type Id": "Bestelltyp-ID",
```
Comment (SirSplasch) [inline lang/de.json:1017]: "Auftragstype-ID"

File: lang/de.json [hunk 33]
```diff
@@ -820,52 +975,69 @@
...
     "Order Type Name": "Name",
     "Order Types": "Auftragsarten",
     "order-confirmation": "Auftragsbestätigung",
+    "Ordering": "Bestellung",
```
Comment (SirSplasch) [inline lang/de.json:1021]: "Sortierung"

File: lang/de.json [hunk 34]
```diff
@@ -907,8 +1084,10 @@
     "Payment Target": "Zahlungsstatus",
     "Payment target": "Zahlungsziel",
     "Payment Target Days": "Zahlungsziel Tage",
+    "Payment Texts": "Zahlungstexte",
     "Payment Type": "Zahlart",
     "Payment type": "Zahlart",
+    "Payment Type Id": "Zahlungstyp-ID",
```
Comment (SirSplasch) [inline lang/de.json:1090]: "Zahlungsart-ID"

File: lang/de.json [hunk 35]
```diff
@@ -944,12 +1129,14 @@
...
     "Product added to cart": "Produkt zum Warenkorb hinzugefügt",
     "Product category discounts": "Produktkategorie Rabatte",
+    "Product Id": "Produkt-ID",
     "Product Image": "Produktbild",
     "Product information": "Produktinformation",
     "Product name, description…": "Produktname, Beschreibung…",
     "Product Number": "Produktnummer",
     "Product number": "Produktnummer",
     "Product Option Group": "Produkt-Option-Gruppe",
+    "Product Option Group Id": "Produktoptionsgruppen-ID",
```
Comment (SirSplasch) [inline lang/de.json:1139]: Produkt-Option-Gruppe-ID

File: lang/de.json [hunk 36]
```diff
@@ -979,10 +1167,17 @@
     "projects.{id}.get": "Projektdetails anzeigen",
     "ProjectTasks": "Projektaufgaben",
     "Protocol": "Protokoll",
+    "Provision": "Provision",
     "Public": "Öffentlich",
     "Purchase Invoice": "Eingangsrechnung",
+    "Purchase Invoice Id": "Kaufrechnung-ID",
```
Comment (SirSplasch) [inline lang/de.json:1173]: "Eingangsrechnung-ID"

File: lang/de.json [hunk 37]
```diff
@@ -979,10 +1167,17 @@
...
+    "Purchase Invoice Positions": "Kaufrechnungspositionen",
```
Comment (SirSplasch) [inline lang/de.json:1174]: "Eingangsrechnungspositionen"

File: lang/de.json [hunk 38]
```diff
@@ -979,10 +1167,17 @@
...
     "Purchase Invoices": "Eingangsrechnungen",
     "Purchase Payment Type": "Einkaufszahlungsart",
+    "Purchase Payment Type Id": "Kaufzahlungstyp-ID",
```
Comment (SirSplasch) [inline lang/de.json:1177]: "Einkaufszahlart-ID"

File: lang/de.json [hunk 39]
```diff
@@ -991,11 +1186,14 @@
     "Queue Monitor": "Warteschlangenmonitor",
     "queued": "In Warteschlange",
     "Quick add": "Schnell hinzufügen",
+    "Rate Percentage": "Satzprozentsatz",
```
Comment (SirSplasch) [inline lang/de.json:1189]: "Satzprozentsatz"???

File: lang/de.json [hunk 40]
```diff
@@ -991,11 +1186,14 @@
...
     "Read terms and conditions": "AGBs und Widerrufsbelehrung lesen",
     "ready_for_delivery": "bereit für Lieferung",
     "ready_for_packing": "bereit zum Packen",
     "Record": "Datensatz",
     "Record new working hours": "Neue Arbeitszeit erfassen",
+    "Recurrences": "Wiederholungen",
+    "Reference Unit Id": "Referenzeinheiten-ID",
```
Comment (SirSplasch) [inline lang/de.json:1196]: "Referenzeinheit-ID"

File: lang/de.json [hunk 41]
```diff
@@ -1007,22 +1205,30 @@
...
+    "Reminder Subject": "Erinnerungsbetreff",
     "Remove": "Entfernen",
     "Repeat": "Wiederhole",
+    "Repeat End": "Wiederholungsende",
     "Repeat end": "Ende Wiederholung",
     "Repeat every": "Wiederhole alle",
     "Repeat password": "Passwort wiederholen",
     "Repeatable": "Wiederholbar",
     "Replicate": "Duplizieren",
+    "Requires Approval": "Genehmigung erforderlich",
```
Comment (SirSplasch) [inline lang/de.json:1220]: "Freigabe erforderlich"

File: lang/de.json [hunk 42]
```diff
@@ -1254,10 +1497,13 @@
     "Uninstall": "Deinstallieren",
     "Unit": "Maßeinheit",
     "Unit Abbreviation": "Einheit Abkürzung",
+    "Unit Gram Weight": "Einheitsgewicht in Gramm",
     "Unit Gross Price": "Einzelpreis Brutto",
+    "Unit Id": "Einheiten-ID",
```
Comment (SirSplasch) [inline lang/de.json:1502]: "Einheit-ID"

File: lang/de.json [hunk 43]
```diff
@@ -1254,10 +1497,13 @@
...
     "Unit Net Price": "Einzelpreis Netto",
     "Unit Price": "Einzelpreis",
     "Unit price :type": "Einzelpreis :type",
+    "Unit Price Price List Id": "Einheitspreisliste-ID",
```
Comment (SirSplasch) [inline lang/de.json:1506]: "Stückpreis-Preisliste-ID"

File: lang/de.json [hunk 44]
```diff
@@ -1384,12 +1635,15 @@
...
     "Variants will be deleted.": "Varianten werden gelöscht.",
+    "Vat Id": "USt-ID",
     "VAT number": "USt-IdNr.",
     "VAT rate": "MwSt.-Satz",
     "Vat Rate": "MwSt.-Satz",
     "Vat rate": "MwSt.-Satz",
+    "Vat Rate Id": "Mehrwertsteuersatz-ID",
     "Vat Rate Percentage": "MwSt.-Satz %",
     "Vat Rates": "MwSt.-Sätze",
+    "Vendor Customer Number": "Lieferantenkundennummer",
```
Comment (SirSplasch) [inline lang/de.json:1646]: can this be removed?

File: lang/de.json [hunk 45]
```diff
@@ -1422,6 +1680,7 @@
     "wire:confirm.uninstall-plugin": "Plugin deinstallieren|Soll das Plugin wirklich deinstalliert werden?|Abbrechen|Deinstallieren",
     "Work Time": "Arbeitszeit",
     "Work Time Type": "Arbeitszeitart",
+    "Work Time Type Id": "Arbeitszeittyp-ID",
```
Comment (SirSplasch) [inline lang/de.json:1683]: "Arbeitszeitart-ID"

File: src/Models/Order.php [hunk 46]
```diff
@@ -478,6 +496,31 @@ public function calculateBalance(): static
         return $this;
     }
 
+    public function calculateMargin(): static
+    {
+        $this->total_purchase_price = $this->orderPositions()
+            ->where('is_alternative', false)
+            ->sum('purchase_price');
+
+        $this->margin = bcround(
```
Comment (SirSplasch) [inline src/Models/Order.php:505]: why not use Rounding Helper?

File: src/Models/Order.php [hunk 47]
```diff
@@ -478,6 +496,31 @@ public function calculateBalance(): static
...
+            2
+        );
+
+        $variableCosts = 0;
+        $variableCosts = bcadd($variableCosts, $this->commissions()->sum('commission'));
+        $variableCosts = bcadd($variableCosts, $this->workTimes()->sum('total_cost'));
+        $variableCosts = bcadd($variableCosts, $this->projects()->sum('total_cost'));
+        $this->total_cost = $variableCosts;
+
+        $this->gross_profit = bcround(
```
Comment (SirSplasch) [inline src/Models/Order.php:516]: why not use Rounding Helper?

File: src/Models/WorkTime.php [hunk 48]
```diff
@@ -63,4 +69,24 @@ public function workTimeType(): BelongsTo
     {
         return $this->belongsTo(WorkTimeType::class);
     }
+
+    public function calculateTotalCost(): static
+    {
+        $this->total_cost = bcmul(
+            $this->user->cost_per_hour,
+            $this->total_time_ms / 1000 / 60 / 60,
```
Comment (SirSplasch) [inline src/Models/WorkTime.php:77]: bcdiv($this->total_time_ms, 3600000)

File: src/Models/WorkTime.php [hunk 49]
```diff
@@ -63,4 +69,24 @@ public function workTimeType(): BelongsTo
...
+            2
+        );
+
+        if ($this->model && method_exists($this->model, 'costColumn') && $this->model->costColumn()) {
```
Comment (SirSplasch) [inline src/Models/WorkTime.php:81]: $costColumn = $this->model->costColumn()
So that this does not need to be called 2 times after.

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: database/migrations/2024_08_01_045824_add_is_purchase_to_price_lists_table.php [hunk 50]
```diff
@@ -0,0 +1,30 @@
...
+
+return new class() extends Migration
+{
+    public function up(): void
+    {
+        Schema::table('price_lists', function (Blueprint $table) {
+            $table->boolean('is_purchase')->default(false)->after('is_default');
+        });
+
+        DB::table('price_lists')->insert([
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply database/migrations/2024_08_01_045824_add_is_purchase_to_price_lists_table.php:16]: to ensure there is a price list for purchase prices for all currently active installations

File: src/Helpers/PriceHelper.php [hunk 51]
```diff
@@ -126,6 +126,7 @@ public function price(): ?Price
         }
 
         if (! $this->price && $this->useDefault) {
+            $this->priceList = PriceList::default();
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply src/Helpers/PriceHelper.php:129]: added resolve static to HasDefault trait instead…

File: src/Models/PriceList.php [hunk 52]
```diff
@@ -28,9 +29,31 @@ protected function casts(): array
             'rounding_method_enum' => RoundingMethodEnum::class,
             'is_net' => 'boolean',
             'is_default' => 'boolean',
+            'is_purchase' => 'boolean',
         ];
     }
 
+    protected static function booted(): void
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply src/Models/PriceList.php:36]: the plan is to have one "default" purchase price, which is same like "standard" flag. Later on you can add more purchase price lists that HAVE to be attached to a supplier. In that case this have to be changed, but i dont plan to show these lists on prices tab

File: src/Models/Product.php [hunk 53]
```diff
@@ -151,10 +151,13 @@ public function productProperties(): BelongsToMany
         );
     }
 
-    public function purchasePrice(float|int $amount): float
+    public function purchasePrice(float|int $amount = 1): Price
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply src/Models/Product.php:154]: for now yes, but when extended purchase prices are added you need to have a amount to calculate the average price depending on the purchase prices that were used for the stock

File: src/Livewire/Forms/OrderPositionForm.php [hunk 54]
```diff
@@ -125,6 +125,7 @@ public function fillFormProduct(?Product $product = null): void
         $this->product_number = $product->product_number;
         $this->ean_code = $product->ean;
         $this->unit_gram_weight = $product->weight_gram;
+        $this->purchase_price = $product->purchasePrice(1)->price;
```
[the remark this answers is not in the document]
Comment (patrickweh) [reply src/Livewire/Forms/OrderPositionForm.php:128]: in line 46

File: lang/de.json (hunk 13 above)
Comment (patrickweh) [reply lang/de.json:680]: thats correct

File: lang/de.json (hunk 19 above)
Comment (patrickweh) [reply lang/de.json:784]: yes, in email settings

File: lang/de.json (hunk 21 above)
Comment (patrickweh) [reply lang/de.json:793]: nope

File: lang/de.json [hunk 55]
```diff
@@ -820,52 +972,69 @@
...
+    "Order Position Id": "Auftragsposition-ID",
     "Order Positions": "Auftragspositionen",
     "Order positions": "Auftragspositionen",
+    "Order saved successfully!": "Auftrag erfolgreich gespeichert!",
     "Order State": "Auftragsstatus",
     "Order state": "Auftragsstatus",
     "Order Type": "Auftragsart",
     "Order type": "Auftragsart",
+    "Order Type Enum": "Auftragstyp",
+    "Order Type Id": "Auftragstyp-ID",
```
Comment (SirSplasch) [inline lang/de.json:1014]: Well, these two should be Auftragsart, my bad.

File: lang/de.json [hunk 56]
```diff
@@ -603,10 +673,21 @@
...
     "Get new messages": "Neue Nachrichten abrufen",
+    "Give": "Geben",
     "Gram": "Gramm",
     "gross": "brutto",
+    "Gross Profit": "Rohertrag",
     "Group": "Gruppe",
+    "Guard Name": "Guard-Name",
+    "Has Delivery Lock": "Hat Liefersperre",
+    "Has Logistic Notify Number": "Hat Versand-Benachrichtigungsnummer",
+    "Has Logistic Notify Phone Number": "Hat Versand-Benachrichtigungsnummer",
```
Comment (SirSplasch) [inline lang/de.json:684]: "Hat Versand-Benachrichtigungstelefonnummer" something like that.

File: src/Models/WorkTime.php [hunk 57]
```diff
@@ -63,4 +69,26 @@ public function workTimeType(): BelongsTo
     {
         return $this->belongsTo(WorkTimeType::class);
     }
+
+    public function calculateTotalCost(): static
+    {
+        $this->total_cost = bcmul(
```
Comment (SirSplasch) [inline src/Models/WorkTime.php:75]: Why not round this? Instead just cut the end?

**Review by SirSplasch — CHANGES_REQUESTED** [review]
Fix failed test

File: src/Models/Product.php (hunk 53 above)
[the remark this answers is not in the document]
Comment (SirSplasch) [reply src/Models/Product.php:154]: make this ?Price for now, otherwise this might throw an exception

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**